File: app/src/main/java/com/example/library2/leetcode/PivotIndex.java
package com.example.library2.leetcode;

public class PivotIndex {
    public static int pivotIndex(int[] nums) {
        int[] sumLeftArray = nums;
        int[] sumRightArray = nums;
        for (int i = 1; i < nums.length; i++) {
            System.out.println("outer loop of left array i="+i+" nums="+nums+" sumLeftArray="+sumLeftArray);
            sumLeftArray[i] += sumLeftArray[i-1];
            for (int j = nums.length - 2; i > 0; i--) {
                System.out.println("inner loop of right array j="+j+" nums="+nums+" sumRightArray="+sumRightArray);
                if (sumRightArray[j] == sumLeftArray[i]) {
                    return i;
                }
                sumRightArray[j] += sumRightArray[j+1];
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        System.out.println(pivotIndex(new int[]{ 1,7,3,6,5,6 }));
    }
}

File: app/src/main/java/com/example/library2/leetcode/PivotIndex.kt
package com.example.library2.leetcode

fun pivotIndex(nums: IntArray): Int {
    var sumLeftArray: IntArray = nums
    var sumRightArray: IntArray = nums
    sumLeftArray[0] = nums[0]
    sumRightArray[nums.size-1] = nums[nums.size-1]
    for (i in 1 until sumLeftArray.size) {
        sumLeftArray[i] += sumLeftArray[i-1]
    }
    for (i in sumRightArray.size-2 downTo  1) {
        sumRightArray[i] += sumRightArray[i-1]
        println(sumRightArray[i])
    }
}

fun main() {
    val nums = intArrayOf(1, 7, 3, 6, 5, 6)
    println(pivotIndex(nums))
}
